File: mySideBar/A03_PatriciaDurango/app/src/main/java/com/example/a03_patriciadurango/MainActivity.java
package com.example.a03_patriciadurango;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Gravity;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private Button botonClick, botonLargo;
    private TextView etiquetaDetalle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        botonClick = findViewById(R.id.btnOnClick);
        botonLargo = findViewById(R.id.btnOnLongClick);
        etiquetaDetalle = findViewById(R.id.txtDetalle);

        configurarEventos();
    }

    private void configurarEventos() {
        botonClick.setOnClickListener(v -> {
            etiquetaDetalle.setText("Presionaste el botón de click normal ✅");
            mostrarToast("Click rápido detectado ⚡", Gravity.TOP);
        });
        botonLargo.setOnLongClickListener(v -> {
            etiquetaDetalle.setText("Dejaste presionado el botón ⏳");
            mostrarToast("Click prolongado registrado 💡", Gravity.CENTER);
            return true;
        });
    }
    private void mostrarToast(String mensaje, int posicion) {
        Toast aviso = Toast.makeText(this, mensaje, Toast.LENGTH_SHORT);
        aviso.setGravity(posicion, 0, 200);
        aviso.show();
    }
}

File: A02_PatriciaDurango4/app/src/main/java/com/example/a02_patriciadurango/MainActivity.kt
package com.example.a02_patriciadurango

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btnRelative = findViewById<Button>(R.id.btnRelativeLayout)
        val btnLinear = findViewById<Button>(R.id.btnLinearLayout)
        val btnTable = findViewById<Button>(R.id.btnTableLayout)
        val btnGrid = findViewById<Button>(R.id.btnGridLayout)

        btnRelative.setOnClickListener {
            startActivity(Intent(this, RelativeLayoutActivity::class.java))
        }

        btnLinear.setOnClickListener {
            startActivity(Intent(this, LinearLayoutActivity::class.java))
        }

        btnTable.setOnClickListener {
            startActivity(Intent(this, TableLayoutActivity::class.java))
        }

        btnGrid.setOnClickListener {
            startActivity(Intent(this, GridLayoutActivity::class.java))
        }
    }
}
